EIPS/eip-7484.md

```diff
@@ -0,0 +1,217 @@
+---
+eip: 7484
+title: Registry Adapters for Smart Accounts
+description: Adapters that allow modular smart contract accounts to verify the security of modules using a Module Registry
+author: Konrad Kopp (@kopy-kat), zeroknots (@zeroknots)
+discussions-to: https://ethereum-magicians.org/t/erc-7484-registry-adapters-for-smart-accounts/15434
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-08-14
+requires: 4337
+---
+
+## Abstract
+
+This proposal standardises a Registry Adapter for modular smart contract accounts. This Adapter allows the account to query and verify security attestations about a module through an Attestation Registry. The adapter is responsible for querying the registry and correctly handling the return values.
+
+## Motivation
+
+[ERC-4337](./eip-4337.md) standardises the execution flow of contract accounts and [ERC-6900](./eip-6900.md) aims to standardise the modular implementation of these accounts, allowing any developer to build modules for these modular accounts (hereafter smart accounts). However, adding third-party modules into smart accounts unchecked opens up a wide range of attack vectors on these accounts.
```

Comment:
Just so you're aware, mentioning ERC-6900 will create an implicit link between this EIP and that one. ERC-7484 won't be able to progress past the status of ERC-6900 (currently Draft.) That's totally fine if that's what you want, I just wanted to point it out.